tests/test_toolchain.py

```diff
@@ -121,7 +122,10 @@ def test_recipes(self):
         Checks the `recipes` command prints out recipes information without crashing.
         """
         argv = ['toolchain.py', 'recipes']
-        with patch_sys_argv(argv), patch_sys_stdout() as m_stdout:
+        with (
+                patch_sys_argv(argv)), (
+                patch_sys_stdout()) as m_stdout, (
+                backup_attributes(Recipe, {'recipes'})):
```

Comment:
in Python3 that could also be used as a decorator, e.g.
```
@backup_attributes(Recipe, {'recipes'})
def test_recipes(self):
    pass
```

Reply:
I'm wondering, will this work when used alongside with `mock.patch` decorator?

Reply:
Depends of what you patch and if you patch before the `backup_attributes` or after I guess

Reply:
Ok, we'll see then, thanks!